geomstats/distributions/lognormal.py

```diff
@@ -0,0 +1,187 @@
+            if metric is None:
+                manifold.metric = SPDMetricLogEuclidean(n)
+            else:
+                metric = manifold.metric
+                if (
+                    not isinstance(metric, SPDMetricLogEuclidean) and
+                    not isinstance(metric, SPDMetricAffine)
+                ):
+                    raise ValueError(
+                        "Invalid Metric, "
+                        "Should be of type SPDMetricLogEuclidean"
+                        "or SPDMetricAffine")
+
+        if cov is not None:
+            if (
+                cov.ndim != 2 or
+                (cov.shape[0], cov.shape[1]) != (cov_n, cov_n)
+            ):
+                valid_shape = (cov_n, cov_n)
+                raise ValueError("Invalid Shape, "
+                                 "cov should have shape", valid_shape)
+
+        else:
+            cov = gs.eye(cov_n)
+
+        self.__manifold = manifold
+        self.__mean = mean
+        self.__cov = cov
+        self.__mean_dim = n
+        self.__cov_dim = cov_n
+
+    @property
+    def manifold(self):
+        return self.__manifold
+
+    @property
+    def mean(self):
+        return self.__mean
+
+    @property
+    def cov(self):
+        return self.__cov
+
+    @property
+    def mean_dim(self):
+        return self.__mean_dim
+
+    @property
+    def cov_dim(self):
+        return self.__cov_dim
+
+    def __sample_spd(self, n_samples):
+        """Generates samples for SPD manifold"""
+        def samples_sym(mean, cov, n_samples):
+            """auxillary function for generating sym matrices"""
+            samples_euclidean = gs.random.multivariate_normal(
+                mean, cov, (n_samples,))
+            diag = samples_euclidean[:, :self.mean_dim]
+            off_diag = samples_euclidean[:, self.mean_dim:] / gs.sqrt(2.0)
+            samples_sym = gs.mat_from_diag_triu_tril(
+                diag=diag, tri_upp=off_diag, tri_low=off_diag)
+            return samples_sym
+
+        if isinstance(self.manifold.metric, SPDMetricLogEuclidean):
+            sym_matrix = self.manifold.logm(self.mean)
+            mean_euclidean = gs.hstack(
+                (gs.diagonal(sym_matrix)[None, :],
+                 gs.sqrt(2.0) * gs.triu_to_vec(sym_matrix, k=1)[None, :]))[0]
+            _samples = samples_sym(mean_euclidean, self.cov, n_samples)
+
+        else:
+            samples_sym = samples_sym(
+                gs.zeros(self.cov_dim), self.cov, n_samples)
+            mean_half = self.manifold.powerm(self.mean, 0.5)
+            _samples = Matrices.mul(mean_half, samples_sym, mean_half)
+
+        return self.manifold.expm(_samples)
+
+    def __sample_euclidean(self, samples):
+        """Generates samples for Euclidean Manifold"""
+        _samples = gs.random.multivariate_normal(
+            self.mean, self.cov, (samples,))
+        return gs.exp(_samples)
+
+    def sample(self, samples=1):
+        """Generates samples"""
```

Comment:
Complete docstring for the non-private method (see our contributing guidelines for details)

Reply:
Generates -> Generate

Reply:
Done